### PR TITLE
I2c s3

### PR DESCRIPTION
**Pull request by ljia5**

### CODE REVIEW

File: drivers/gpu/drm/bridge/ti/fpd_dp_ser_drv.c [hunk 1]
```diff
@@ -183,15 +184,23 @@ int fpd_dp_ser_write_reg(struct i2c_client *client, unsigned int reg_addr, u8 va
...
-		fpd_dp_ser_debug("[FPD_DP] [-%s-%s-%d-], fail client->addr=0x%02x, reg_addr=0x%02x, val=0x%02x\n",
-				__FILE__, __func__, __LINE__, client->addr, reg_addr, val);
+	while ((ret != 0) && (count < 10)) {
+		ret = i2c_transfer(client->adapter, &msg, 1);
+		if (ret < 0) {
+			count++;
+			fpd_dp_ser_debug("[FPD_DP] [-%s-%s-%d-], fail client->addr=0x%02x, reg_addr=0x%02x, val=0x%02x\n",
+					__FILE__, __func__, __LINE__, client->addr, reg_addr, val);
+		} else
+			ret	= 0;
```
Comment (phreer) [inline drivers/gpu/drm/bridge/ti/fpd_dp_ser_drv.c:194]: Better to format the code consistently.

File: drivers/gpu/drm/bridge/ti/fpd_dp_ser_drv.c [hunk 2]
```diff
@@ -2134,14 +2187,12 @@ bool fpd_dp_ser_init(void)
...
 
-	WRITE_ONCE(deser_reset, 0);
-
 	fpd_dp_ser_set_up_mcu(fpd_dp_priv->priv_dp_client[0]);
 
-	fpd_dp_ser_set_ready(true);
 	fpd_dp_ser_unlock_global();
 
-	queue_work(fpd_dp_priv->motor_setup_wq, &fpd_dp_priv->motor_setup_work);
+	if (wait_display_startup_done(fpd_dp_priv->priv_dp_client[2]))
```
Comment (phreer) [inline drivers/gpu/drm/bridge/ti/fpd_dp_ser_drv.c:2194]: Function fpd_dp_ser_init is expected to return false when it fails to initialize serdes.

File: drivers/input/touchscreen/ef1e_tp/ef1e_tp_driver.c [hunk 3]
```diff
@@ -641,6 +641,29 @@ static int ef1e_tp_remove(struct platform_device *dev)
 	return 0;
 }
 
+static void ef1e_dp_shutdown(struct platform_device *pdev)
```
Comment (phreer) [inline drivers/input/touchscreen/ef1e_tp/ef1e_tp_driver.c:644]: The resources would be double-freed here as they have been released in the `.remove` callback.

What's the purpose of preferrence of `.shutdown` over `.remove`?

**Review by phreer — COMMENTED** [review]
Does this PR fix the issue when doing cold boot?

File: drivers/input/touchscreen/ef1e_tp/ef1e_tp_driver.c (hunk 3 above)
Comment (ljia5) [reply drivers/input/touchscreen/ef1e_tp/ef1e_tp_driver.c:645]: Remove free resources. S5 stability test call shutdown, don`t call remove.

**Comment by ljia5** [comment]
pr 134 included this pr

**Comment by ljia5** [comment]
close error

**Review by phreer — APPROVED** [review]
Looks good to me.

**Comment by feijiang1** [comment]
It is better you run scripts/checkpatch.pl to verify your patch doesn't have coding style problem.

**Review by feijiang1 — APPROVED** [review]
LGTM

**Pull request merged**